### DISCUSSION TITLE
Question about access to single pixels

### DISCUSSION THREAD

**Discussion by ewelot** [libvips/libvips#2270] [Q&A] [answered]
It's probably a rather basic question but I have not been able to find a quick solution by myself ...
What I like to do is gathering single pixels (pels) matching a condition set on their value and print out both image coordinates and the value. I figured (using pyvips) how to read the image and build a mask according the condition but I'd like to get an advice on how to print all matching pixels (x, y, value). Shall I use numpy functionality or can this be done in pyvips directly?

One possible application is creating a simple text file of bad pixels for later use by _dcraw_ with -P option.

Thanks for help,
Thomas

**Comment by jcupitt** [comment, marked as answer]
Hi @ewelot,

You can use `max` (or `min`) to find arrays of points. The syntax is a bit awkward, but eg.:

```python
#!/usr/bin/python3
  
import sys
import pyvips

image = pyvips.Image.new_from_file(sys.argv[1])
mask = image[0] == 42

[value, opts] = mask.max(size=100, x_array=True, y_array=True)
x = opts["x_array"]
y = opts["y_array"]

for i in range(len(x)):
    print(f"{x[i]}, {y[i]}")
```

Will find the first 100 pixels with 42 in the red channel.

**Reply by ewelot** [reply]
Great, that's exactly what I was looking for!
